fennel/featuresets/featureset.py

```diff
@@ -311,9 +319,108 @@ def __eq__(self, other: Any) -> bool:
     def __str__(self) -> str:
         return self.name
 
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, name: str) -> None:
+        self._name = name
+        if self.extractor:
+            self.extractor.output_features = [name]
+
+    @property
+    def featureset_name(self) -> str:
+        return self._featureset_name
+
+    @featureset_name.setter
+    def featureset_name(self, name: str) -> None:
+        self._featureset_name = name
+        if self.extractor:
+            self.extractor.featureset = name
+
     def fqn(self) -> str:
         return self.fqn_
 
+    def extract(
+        self,
+        field: Field = None,
+        provider: Featureset = None,
+        default=None,
+        feature: Feature = None,
+        version: int = 0,
+    ) -> Feature:
+        """
+        Derives an extractor for the feature using the given params.
+        The derived extractor either performs a lookup on the given field,
+        or is an alias to the given feature.
+
+        Parameters:
+        field: the field in a dataset to lookup and return
+        provider: the input featureset that should contain a feature matching
```

Comment:
matching the key field names, and there could be multiple of them. So please fix this accordingly